test/libdnf/utils.hpp

```diff
@@ -93,6 +97,70 @@ struct assertion_traits<libdnf::advisory::AdvisorySet> {
     }
 };
 
+template <>
+struct assertion_traits<libdnf::comps::Environment> {
+    inline static bool equal(const libdnf::comps::Environment & left, const libdnf::comps::Environment & right) {
+        return left == right;
+    }
+
+    inline static std::string toString(const libdnf::comps::Environment & environment) {
+        std::string repos;
+        for (const auto & repo : environment.get_repos()) {
+            if (!repos.empty()) {
+                repos += ", ";
+            }
+            repos += repo;
```

Comment:
This is not great, but not sure how to do it better, without the `if` part. Also, I assume this method is just to uniquely describe an Environment, so that's why I am adding the id and repos to the string.

Reply:
I think this is just fine. Yes, the purpose is to print all important data of the object in the unittest output. You can add more than the necessary unique identifier if you think it'd be useful and won't add too much clutter. In this case the id and the list of repos seems good.